src/sympc/protocol/falcon/falcon.py

```diff
@@ -451,3 +454,87 @@ def select_shares(x: MPCTensor, y: MPCTensor, b: MPCTensor) -> MPCTensor:
         z = x + (d * (y - x))
 
         return z
+
+    @staticmethod
+    def _random_prime_group(
+        session: Session, shape: Union[torch.Size, tuple]
+    ) -> MPCTensor:
+        """Computes shares of random number in Zp*.Zp* is the multiplicative group mod p.
+
+        Args:
+            session (Session): session to generate random shares for.
+            shape (Union[torch.Size,tuple]): shape of the random share to generate.
+
+        Returns:
+            share (MPCTensor): Retuns shares of random number in group Zp*.
+
+        Zp* = {1,2..,p-1},where p is a prime number.
+        We use Euler's Theorum for verifying that random share is not zero.
+        It states that:
+        For a general modulus n
+        a^phi(n) = 1(mod n), if a is co prime to n.
+        In our case n=p(prime number), phi(p) = p-1
+        phi(n) = Euler totient function.
+        """
+        while True:
+            ptr_list: List[ReplicatedSharedTensor] = []
+            for session_ptr in session.session_ptrs:
+                ptr = session_ptr.prrs_generate_random_share(
+                    shape=(), ring_size=str(PRIME_NUMBER)
+                ).resolve_pointer_type()
```

Comment:
Q: Can't we remove the resolve pointer here?

Reply:
It returns an `Any Pointer` after `repeat` method,which could not be used in subsequent operations, so I used `resolve_pointer_type`

Reply:
But it should be good I think. `Any` pointer is only for us to know what methods are on that pointer, but you should be able to still use that pointer I think.
Q: Could you remove the `resolve_pointer_type` and check if it works without it?

Reply:
I tried removing it george ,I got an error ,as we subscript share ptrs in several places of the underlying tensor.

Reply:
Also that I think PRRS has a Union of RSTensor and ShareTensor in API.py so typically such methods require resolve_pointer_type().

Reply:
George, in Any pointer , the tensor specific methods are not copied I think , which resulted in error at Falcon multiplication, which used tensor specific methods of RSTensor.